.github/workflows/workflow.yml

```diff
@@ -0,0 +1,44 @@
+name: workflow
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  # Reference: https://github.com/golangci/golangci-lint-action
+  golangci-lint:
+    name: linting
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.19'
+          cache: false
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.53.3
+
+  go-test:
+    needs: [golangci-lint]
```

Comment:
In case we define needs, this job will not run before the lint job runs.
What is de down side here? We will start a new (fresh) machine for that.
Is that desired? Can't we simply run lining inside the other job after the testing step?

Reply:
Well first of all thinking how the low level infrastructure would execute this should not impact how we think about CI at higher level. But to answer I don't have much idea about github's CI behavior. Having said that here is what we do internally: We  run basic steps of linting and unit tests as git pre-push hook in local dev machines. This way we make sure that these steps will 'almost' never fail is CI plus we can have these steps run in parallel knowing that there is little chance of some of the steps going to waste if one fails.

Even if we we do not use pre-push etc., we have to see this case-by-case basis. This repo and Go code in general are very 'light' on resources hence overall if some jobs fail and go to waste occasionally then it is okay IMO.

Earlier I added dependencies but after thinking more deeply to answer your query I think going full parallel is okay for this repo. Of course we can change this later.